spec/models/piece_spec.rb

```diff
@@ -5,19 +5,76 @@
   let(:white_player) { FactoryGirl.create(:player, email: 'blah@blah.com', password: 'SPACECAT') }
   let(:black_player) { FactoryGirl.create(:player, email: 'meow@meow.com', password: 'MONORAILCAT') }
   let(:game) { FactoryGirl.create(:game, white_player_id: white_player.id, black_player_id: black_player.id) }
-  let(:piece) { FactoryGirl.create(:piece, x_position: 4, y_position: 4, game_id: game.id, player_id: white_player.id) }
 
-  let(:pawn) { FactoryGirl.create(:pawn, x_position: 2, y_position: 1, game_id: game.id, player_id: black_player.id) }
+  describe "King#valid_move?" do
+
+    let(:king) { FactoryGirl.create(:king, x_position: 4, y_position: 4, game_id: game.id, player_id: black_player.id) }
+    
+    subject { king.valid_move?(destination_x, destination_y) }
+
+    context "valid move" do
+      let(:destination_x) { 5 }
+      let(:destination_y) { 5 }
+
+      it "should return true if the move is valid" do
+        expect(subject).to eq(true)
+      end
+    end
+
+    context "invalid move" do
+      let(:destination_x) { 7 }
+      let(:destination_y) { 7 }
+
```

Comment:
The super call is untested. I would recommend trying to move the piece on top of itself for that.